agent/cache/cache.go

```diff
@@ -723,6 +723,22 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 		// Set our entry
 		c.entriesLock.Lock()
 
+		if _, ok := c.entries[key]; !ok {
```

Comment:
I wonder if it is possible for a new entry to be added to the `entries` map while the fetch is happening. In that case we would still leak a goroutine right? We could compare the entry here, maybe by using the `Expiry` which is a pointer.

```suggestion
		if cachedEntry, ok := c.entries[key]; !ok || cachedEntry.Expiry != entry.Expiry {
```

Probably not a big deal since the follow up PR makes this effectively impossible.

Reply:
🤔 I _think_ if we get a double race and the entry expires and then is legitimately recreated by a new client `Get` during fetch then the behavior is still correct (i.e. not leaky) because by definition the client that recreated the entry is still interested in keeping it alive right? So it's not really being leaked in that case but rather kept because there is a client that cares about it still?

But maybe I'm missing a possible edge case?